shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import android.view.textservice.SentenceSuggestionsInfo;
+import android.view.textservice.SpellCheckerSession;
+import android.view.textservice.SuggestionsInfo;
+import android.view.textservice.TextInfo;
+import android.view.textservice.TextServicesManager;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.SpellCheckChannel;
+import io.flutter.plugin.localization.LocalizationPlugin;
+import java.util.ArrayList;
+import java.util.Locale;
+
+/**
+ * {@link SpellCheckPlugin} is the implementation of all functionality needed for spell check for
+ * text input.
+ *
+ * <p>The plugin handles requests for spell check sent by the {@link
+ * io.flutter.embedding.engine.systemchannels.SpellCheckChannel} via sending requests to the Android
+ * spell checker. It also receive the spell check results from the service and send them back to the
+ * framework through the {@link io.flutter.embedding.engine.systemchannels.SpellCheckChannel}.
+ */
+public class SpellCheckPlugin
+    implements SpellCheckChannel.SpellCheckMethodHandler,
+        SpellCheckerSession.SpellCheckerSessionListener {
+
+  private final SpellCheckChannel mSpellCheckChannel;
+  private final TextServicesManager mTextServicesManager;
+  private SpellCheckerSession mSpellCheckerSession;
+
+  // The maximum number of suggestions that the Android spell check service is allowed to provide
+  // per word.
+  // Same number that is used by default for Android's TextViews.
+  private static final int MAX_SPELL_CHECK_SUGGESTIONS = 5;
```

Comment:
If we move this configuration to the framework, will it affects the performance on Android?

Maybe consider adding a similar configuration in framework to unify the experience on all platforms.

Reply:
This would mean we would have to send this number to the engine and versus just having it live in this plugin that it's directly tied to, so I feel like it would be best to leave it here unless all platforms need this parameter. It doesn't look like iOS has a similar parameter, but correct me if I'm wrong.